### ISSUE TITLE
about resume training, loss is large as initial

### ISSUE THREAD

**Issue by lzed2399** [open-mmlab/mmdetection#7487]
I use 'resume_from = xxx.pth ' to continue training, and find the loss is as larger as initial, mAP also decrease. How to solve it ?  Many thanks!

**Comment by Czm369** [comment]
Maybe you can check the path of checkpoint.

**Comment by HajunKim** [comment]
That's what I happening to me as well.  
not exactly as large as the initial but still way more larger than the loss I got at the last.

**Comment by liuqunzhong** [comment]
Hello, I encountered a similar problem when resume training.

I train a yolov7 face detector, config is modified on the “yolov7/yolov7_l_syncbn_fast_8x16b-300e_coco.py”, training data is our customized face dataset.
Training is Interrupted after save epoch_37.pth, because ssd disk is full.
I resume training from 37, explicit specify the resume checkpoint config:
load_from = '/home/share/train_schedule/xxxxx/yolov7_l_640_640_face/epoch_37.pth'
resume = True

Successfully resumed from 37, and beginning 38 epoch.
In resume log, there is a warning:
"WARNING - resume_param_scheduler is True but self.param_schedulers is None, so skip resuming parameter schedulers"
And the loss is abnormal jumping from 37 to 38, In 37 epoch，loss is 0.2 ~ 0.23, In 38 epoch, loss is 0.29 ~ 0.33. The two logs are pasted later.
@hhaAndroid
Is this a normal phenomenon?
Is it related to the checkpoint_config " save_param_scheduler=False"?

###############https://github.com/open-mmlab/mmyolo/pull/37####################
2023/07/09 03:20:54 - mmengine - INFO - Epoch(train) [37][7100/8322] lr: 6.9758e-05 eta: 1 day, 13:32:14 time: 0.4040 data_time: 0.0186 memory: 8371 loss: 0.2353 loss_cls: 0.0000 loss_obj: 0.0430 loss_bbox: 0.1923
2023/07/09 03:21:34 - mmengine - INFO - Epoch(train) [37][7200/8322] lr: 6.9758e-05 eta: 1 day, 13:31:38 time: 0.3842 data_time: 0.0183 memory: 8380 loss: 0.2191 loss_cls: 0.0000 loss_obj: 0.0396 loss_bbox: 0.1794
2023/07/09 03:22:13 - mmengine - INFO - Epoch(train) [37][7300/8322] lr: 6.9758e-05 eta: 1 day, 13:31:02 time: 0.3890 data_time: 0.0180 memory: 8388 loss: 0.2212 loss_cls: 0.0000 loss_obj: 0.0350 loss_bbox: 0.1862
2023/07/09 03:22:52 - mmengine - INFO - Epoch(train) [37][7400/8322] lr: 6.9758e-05 eta: 1 day, 13:30:27 time: 0.3754 data_time: 0.0179 memory: 8380 loss: 0.2095 loss_cls: 0.0000 loss_obj: 0.0335 loss_bbox: 0.1760
2023/07/09 03:22:55 - mmengine - INFO - Exp name: yolov7_l_640_640_face_1_20230707_183430
2023/07/09 03:23:31 - mmengine - INFO - Epoch(train) [37][7500/8322] lr: 6.9758e-05 eta: 1 day, 13:29:50 time: 0.3892 data_time: 0.0180 memory: 8383 loss: 0.2055 loss_cls: 0.0000 loss_obj: 0.0392 loss_bbox: 0.1663
2023/07/09 03:24:10 - mmengine - INFO - Epoch(train) [37][7600/8322] lr: 6.9758e-05 eta: 1 day, 13:29:14 time: 0.3876 data_time: 0.0179 memory: 8371 loss: 0.1986 loss_cls: 0.0000 loss_obj: 0.0330 loss_bbox: 0.1657
2023/07/09 03:24:50 - mmengine - INFO - Epoch(train) [37][7700/8322] lr: 6.9758e-05 eta: 1 day, 13:28:38 time: 0.3911 data_time: 0.0181 memory: 8371 loss: 0.2236 loss_cls: 0.0000 loss_obj: 0.0420 loss_bbox: 0.1816
2023/07/09 03:25:29 - mmengine - INFO - Epoch(train) [37][7800/8322] lr: 6.9758e-05 eta: 1 day, 13:28:02 time: 0.3856 data_time: 0.0183 memory: 8371 loss: 0.2211 loss_cls: 0.0000 loss_obj: 0.0426 loss_bbox: 0.1785
2023/07/09 03:26:09 - mmengine - INFO - Epoch(train) [37][7900/8322] lr: 6.9758e-05 eta: 1 day, 13:27:27 time: 0.3865 data_time: 0.0184 memory: 8410 loss: 0.2210 loss_cls: 0.0000 loss_obj: 0.0432 loss_bbox: 0.1778
2023/07/09 03:26:50 - mmengine - INFO - Epoch(train) [37][8000/8322] lr: 6.9758e-05 eta: 1 day, 13:26:52 time: 0.3910 data_time: 0.0186 memory: 8443 loss: 0.2114 loss_cls: 0.0000 loss_obj: 0.0333 loss_bbox: 0.1781
2023/07/09 03:27:28 - mmengine - INFO - Epoch(train) [37][8100/8322] lr: 6.9758e-05 eta: 1 day, 13:26:15 time: 0.3856 data_time: 0.0187 memory: 8373 loss: 0.2035 loss_cls: 0.0000 loss_obj: 0.0334 loss_bbox: 0.1701
2023/07/09 03:28:07 - mmengine - INFO - Epoch(train) [37][8200/8322] lr: 6.9758e-05 eta: 1 day, 13:25:38 time: 0.3812 data_time: 0.0180 memory: 8380 loss: 0.2277 loss_cls: 0.0000 loss_obj: 0.0430 loss_bbox: 0.1846
2023/07/09 03:28:42 - mmengine - INFO - Epoch(train) [37][8300/8322] lr: 6.9758e-05 eta: 1 day, 13:24:59 time: 0.3589 data_time: 0.0163 memory: 8371 loss: 0.2211 loss_cls: 0.0000 loss_obj: 0.0446 loss_bbox: 0.1765
2023/07/09 03:28:50 - mmengine - INFO - Exp name: yolov7_l_640_640_face_1_20230707_183430
2023/07/09 03:28:54 - mmengine - INFO - Saving checkpoint at 37 epochs

###############https://github.com/open-mmlab/mmyolo/pull/38 resume####################
2023/07/10 14:20:16 - mmengine - INFO - Load checkpoint from /home/share/train_schedule/xxxxx/yolov7_l_640_640_face/epoch_37.pth
2023/07/10 14:20:16 - mmengine - WARNING - resume_param_scheduler is True but self.param_schedulers is None, so skip resuming parameter schedulers
2023/07/10 14:20:16 - mmengine - INFO - resumed epoch: 37, iter: 307914
2023/07/10 14:20:16 - mmengine - WARNING - "FileClient" will be deprecated in future. Please use io functions in https://mmengine.readthedocs.io/en/latest/api/fileio.html#file-io
2023/07/10 14:20:16 - mmengine - WARNING - "HardDiskBackend" is the alias of "LocalBackend" and the former will be deprecated in future.
2023/07/10 14:20:16 - mmengine - INFO - Checkpoints will be saved to /home/share/train_schedule/xxxxx/yolov7_l_640_640_face.
2023/07/10 14:20:53 - mmengine - INFO - Exp name: yolov7_l_640_640_face_1_20230710_141823
2023/07/10 14:20:58 - mmengine - INFO - Epoch(train) [38][ 100/8322] lr: 6.7856e-05 eta: 1 day, 18:20:56 time: 0.3853 data_time: 0.0171 memory: 8972 loss: 0.2996 loss_cls: 0.0000 loss_obj: 0.0486 loss_bbox: 0.2511
2023/07/10 14:21:43 - mmengine - INFO - Epoch(train) [38][ 200/8322] lr: 6.7856e-05 eta: 1 day, 19:14:18 time: 0.4389 data_time: 0.0227 memory: 8405 loss: 0.3170 loss_cls: 0.0000 loss_obj: 0.0577 loss_bbox: 0.2593
2023/07/10 14:22:26 - mmengine - INFO - Epoch(train) [38][ 300/8322] lr: 6.7856e-05 eta: 1 day, 18:58:39 time: 0.4310 data_time: 0.0215 memory: 8384 loss: 0.3154 loss_cls: 0.0000 loss_obj: 0.0515 loss_bbox: 0.2639
2023/07/10 14:23:10 - mmengine - INFO - Epoch(train) [38][ 400/8322] lr: 6.7856e-05 eta: 1 day, 19:07:08 time: 0.4417 data_time: 0.0245 memory: 8387 loss: 0.3259 loss_cls: 0.0000 loss_obj: 0.0511 loss_bbox: 0.2747
2023/07/10 14:23:55 - mmengine - INFO - Epoch(train) [38][ 500/8322] lr: 6.7856e-05 eta: 1 day, 19:15:26 time: 0.4342 data_time: 0.0225 memory: 8384 loss: 0.3110 loss_cls: 0.0000 loss_obj: 0.0537 loss_bbox: 0.2573
2023/07/10 14:24:39 - mmengine - INFO - Epoch(train) [38][ 600/8322] lr: 6.7856e-05 eta: 1 day, 19:21:02 time: 0.4343 data_time: 0.0223 memory: 8395 loss: 0.3297 loss_cls: 0.0000 loss_obj: 0.0536 loss_bbox: 0.2762
2023/07/10 14:25:22 - mmengine - INFO - Epoch(train) [38][ 700/8322] lr: 6.7856e-05 eta: 1 day, 19:11:32 time: 0.4280 data_time: 0.0231 memory: 8477 loss: 0.3235 loss_cls: 0.0000 loss_obj: 0.0508 loss_bbox: 0.2726
2023/07/10 14:26:04 - mmengine - INFO - Epoch(train) [38][ 800/8322] lr: 6.7856e-05 eta: 1 day, 18:57:26 time: 0.4125 data_time: 0.0199 memory: 8390 loss: 0.2927 loss_cls: 0.0000 loss_obj: 0.0413 loss_bbox: 0.2514
2023/07/10 14:26:45 - mmengine - INFO - Epoch(train) [38][ 900/8322] lr: 6.7856e-05 eta: 1 day, 18:37:01 time: 0.3938 data_time: 0.0181 memory: 8396 loss: 0.3010 loss_cls: 0.0000 loss_obj: 0.0492 loss_bbox: 0.2519
2023/07/10 14:27:29 - mmengine - INFO - Epoch(train) [38][1000/8322] lr: 6.7856e-05 eta: 1 day, 18:42:05 time: 0.4396 data_time: 0.0235 memory: 8384 loss: 0.3207 loss_cls: 0.0000 loss_obj: 0.0516 loss_bbox: 0.2691

> That's what I happening to me as well. not exactly as large as the initial but still way more larger than the loss I got at the last.

**Comment by iacopo97** [comment]
Has anyone solve the issue?

**Comment by iacopo97** [comment]
The problem could be related to the random state, I suggest you to store the state of randomness and when resuming the training load also the random state that you had when the training failed.

**Comment by 971022jing** [comment]
> That's what I happening to me as well. not exactly as large as the initial but still way more larger than the loss I got at the last.

Hello！ Have you solved this problem?
